src/parser/parser.go

```diff
@@ -0,0 +1,1228 @@
+			Right:      right,
+			NodeCommon: nodes.NodeCommon{Lineno: lineno},
+		}
+		lineno = p.stream.Current().Lineno
+	}
+	return left, nil
+}
+
+func (p *parser) parseAnd() (nodes.Node, error) {
+	lineno := p.stream.Current().Lineno
+	left, err := p.parseNot()
+	if err != nil {
+		return nil, err
+	}
+	for p.stream.SkipIf("name:and") {
+		right, err := p.parseNot()
+		if err != nil {
+			return nil, err
+		}
+		left = &nodes.And{
+			Left:       left,
+			Right:      right,
+			NodeCommon: nodes.NodeCommon{Lineno: lineno},
+		}
+		lineno = p.stream.Current().Lineno
+	}
+	return left, nil
+}
+
+func (p *parser) parseNot() (nodes.Node, error) {
+	if p.stream.Current().Test("name:not") {
+		lineno := p.stream.Next().Lineno
+		n, err := p.parseNot()
+		if err != nil {
+			return nil, err
+		}
+		return &nodes.Not{
+			Node:       n,
+			NodeCommon: nodes.NodeCommon{Lineno: lineno},
+		}, nil
+	}
+	return p.parseCompare()
+}
+
+func (p *parser) parseCompare() (nodes.Node, error) {
+	lineno := p.stream.Current().Lineno
+	expr, err := p.parseMath1()
+	if err != nil {
+		return nil, err
+	}
+	var ops []nodes.Operand
+
+	addOperand := func(tokenType string) error {
+		e, err := p.parseMath1()
+		if err != nil {
+			return err
+		}
+		ops = append(ops, nodes.Operand{
+			Op:         tokenType,
+			Expr:       e,
+			NodeCommon: nodes.NodeCommon{Lineno: lineno},
+		})
+		return nil
+	}
+
+	for {
+		tokenType := p.stream.Current().Type
+		if compareOperators.Has(tokenType) {
+			p.stream.Next()
+			if err := addOperand(tokenType); err != nil {
+				return nil, err
+			}
+		} else if p.stream.SkipIf("name:in") {
+			if err := addOperand("in"); err != nil {
+				return nil, err
+			}
+		} else if p.stream.Current().Test("name:not") && p.stream.Look().Test("name:in") {
+			p.stream.Skip(2)
+			if err := addOperand("notin"); err != nil {
+				return nil, err
+			}
+		} else {
+			break
+		}
+		lineno = p.stream.Current().Lineno
+	}
+
+	if len(ops) == 0 {
+		return expr, nil
+	}
+	return &nodes.Compare{
+		Expr:       expr,
+		Ops:        ops,
+		NodeCommon: nodes.NodeCommon{Lineno: lineno},
+	}, nil
+}
+
+func (p *parser) parseMath1() (nodes.Node, error) {
+	lineno := p.stream.Current().Lineno
+	left, err := p.parseConcat()
+	if err != nil {
+		return nil, err
+	}
+
+	for p.stream.Current().Type == lexer.TokenAdd || p.stream.Current().Type == lexer.TokenSub {
+		currentType := p.stream.Current().Type
+		p.stream.Next()
+		right, err := p.parseConcat()
+		if err != nil {
+			return nil, err
+		}
+		left = makeMathNode(left, right, currentType, lineno)
+		lineno = p.stream.Current().Lineno
+	}
+	return left, nil
+}
+
+func (p *parser) parseMath2() (nodes.Node, error) {
+	lineno := p.stream.Current().Lineno
+	left, err := p.parsePow()
+	if err != nil {
+		return nil, err
+	}
+
+	for p.stream.Current().Type == lexer.TokenMul || p.stream.Current().Type == lexer.TokenDiv || p.stream.Current().Type == lexer.TokenFloordiv || p.stream.Current().Type == lexer.TokenMod {
```

Comment:
can be changed to `slices.COntains([]string{alltokens}, p.stream.Current().Type)`